### PR TITLE
[shared_preferences_foundation] Add Swift runtime search paths for Objective-C apps

### PR DESCRIPTION
**Pull request by jmagman**
`shared_preferences_foundation` is using Swift as of https://github.com/flutter/plugins/pull/6940.  The Objective-C projects do not automatically set up the search paths to find the Swift runtimes.  Hook that up in the `shared_preferences_foundation` podspec.

See explanation at https://github.com/flutter/flutter/issues/16049#issuecomment-552060349 and the first time this was done in plugins: https://github.com/flutter/plugins/pull/6369

Convert the example project back to Objective-C to catch future versions of this issue.

Fixes https://github.com/flutter/flutter/issues/118418

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran [the auto-formatter]. (Unlike the flutter/flutter repo, the flutter/plugins repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/plugins/blob/main/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/plugins/blob/main/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates
[following repository CHANGELOG style]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changelog-style
[the auto-formatter]: https://github.com/flutter/plugins/blob/main/script/tool/README.md#format-code
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests

### CODE REVIEW

File: packages/shared_preferences/shared_preferences_foundation/darwin/shared_preferences_foundation.podspec
Comment (jmagman) [inline packages/shared_preferences/shared_preferences_foundation/darwin/shared_preferences_foundation.podspec:23]: This is the fix.

File: packages/shared_preferences/shared_preferences_foundation/example/ios/RunnerTests/RunnerTests.m
Comment (jmagman) [inline packages/shared_preferences/shared_preferences_foundation/example/ios/RunnerTests/RunnerTests.m:14]: This test is pretty lame, I didn't re-write the better Swift tests back to Objective-C.  Will do that after this lands, to unblock the tree ASAP.

Comment (stuartmorgan) [reply packages/shared_preferences/shared_preferences_foundation/example/ios/RunnerTests/RunnerTests.m:14]: Do we have to abandon the Swift tests? I would have thought we could have Swift tests even though the app is Objective-C.

**Review by stuartmorgan — APPROVED** [review]
LGTM to fix the regression; one question about the test follow-up.

**Comment by stuartmorgan** [comment]
(Thanks for cleaning this up, and sorry I ignored the warning sign of the CI issue when I switched the app-facing package.)

File: packages/shared_preferences/shared_preferences_foundation/example/ios/RunnerTests/RunnerTests.m
Comment (jmagman) [reply packages/shared_preferences/shared_preferences_foundation/example/ios/RunnerTests/RunnerTests.m:14]: I think that does the automagical thing with the Swift runtimes when running an XCTest since Xcode does all kinds of voodoo to get the tests to see non-public APIs?  Relying on this from the example app seems fragile though, it's just luck that we caught it in the storekit example app.  I'll try to think about a better way to holistically test this problem...

**Comment by jmagman** [comment]
Merging. @stuartmorgan to publish tomorrow, which should green out the tree.

**Pull request merged**
